### ISSUE TITLE
Upgrading chart from v0.5.0 to v0.6.0 resource conflicts

### ISSUE THREAD

**Issue by popopanda** [hashicorp/vault-helm#324]
Hello,

Trying to upgrade our vault helm chart from v0.5.0 to v0.6.0.

V0.5.0 was installed via the helm repository

Receiving the following error:
```
$ helm3 -n vault upgrade vault ./ -f values-dev.yaml
Error: UPGRADE FAILED: rendered manifests contain a new resource that already exists. Unable to continue with update: existing resource conflict: namespace: vault, name: vault, existing_kind: networking.k8s.io/v1beta1, Kind=Ingress, new_kind: networking.k8s.io/v1beta1, Kind=Ingress
```

```
$ helm3 ls -n vault
NAME 	NAMESPACE	REVISION	UPDATED                             	STATUS  	CHART      	APP VERSION
vault	vault    	20      	2020-06-04 16:00:56.862093 -0700 PDT	deployed	vault-0.5.0
```

**Comment by tvoran** [comment]
Hi @popopanda, vault-helm 0.5.0 installed an Ingress object with apiVersion of `extensions/v1beta1`, and vault-helm 0.6.0 will use apiVersion: `networking.k8s.io/v1beta1` if available. Since the error mentions the resource already exists, I wonder if there was an Ingress object created manually at some point?

When I try helm upgrading from 0.5.0 to 0.6.0 on k8s 1.18 and 1.15 it is successful, so I'd be curious to know what version of kubernetes you're working with. I'm also using helm 3.2.1.

**Comment by popopanda** [comment]
I do have an ingress with annotations that configures ALB via the ALB ingress controller 

```
  ingress:
    enabled: true
    labels: {}
      # traffic: external
    annotations: |
      kubernetes.io/ingress.class: alb
      alb.ingress.kubernetes.io/scheme: internal
      alb.ingress.kubernetes.io/tags: <removed>
      alb.ingress.kubernetes.io/inbound-cidrs: <removed>
      alb.ingress.kubernetes.io/certificate-arn: 	<removed>
      alb.ingress.kubernetes.io/ssl-policy: <removed>
      alb.ingress.kubernetes.io/listen-ports: '[{"HTTPS": 443}]'
      alb.ingress.kubernetes.io/healthcheck-path: /v1/sys/health?standbyok=true
      alb.ingress.kubernetes.io/healthcheck-protocol: 'HTTPS'
      alb.ingress.kubernetes.io/backend-protocol: 'HTTPS'
```

I am running on EKS v1.15, helm v3.1.1

**Comment by tvoran** [comment]
This seems to be an issue that is fixed in helm v3.2 and later: https://github.com/helm/helm/issues/7697#issuecomment-613535044

If you need to stay on helm v3.1.1, you can delete the ingress object, then run the helm upgrade.

Otherwise you can annotate the ingress object with the new helm annotations, then run the upgrade with helm 3.2.

Example annotations for helm release 'vault' in the vault namespace:
```
kubectl annotate ingress vault meta.helm.sh/release-name=vault -n vault
kubectl annotate ingress vault meta.helm.sh/release-namespace=vault -n vault
kubectl label ingress vault app.kubernetes.io/managed-by=Helm -n vault
```

**Comment by popopanda** [comment]
Spectacular, upgrading to helm 3.2 and annotating my ingress as you suggested resolved my problem

Thanks
